tests/flow/test_svs_tiered.py

```diff
@@ -0,0 +1,605 @@
+    # Search knn in svs.
+    svs_labels, svs_dist = svs_index.knn_query(query_data, k)
+
+    # Compare.
+    has_diff = False
+    for i, svs_res_label in enumerate(svs_labels[0]):
+        if svs_res_label != tiered_labels[0][i]:
+            has_diff = True
+            test_logger.info(f"svs label = {svs_res_label}, tiered label = {tiered_labels[0][i]}")
+            test_logger.info(f"svs dist = {svs_dist[0][i]}, tiered dist = {tiered_dist[0][i]}")
+
+    assert not has_diff
+    test_logger.info(f"svs graph is identical to the tiered index graph")
+
+
+def test_recall_after_deletion(test_logger):
+
+    indices_ctx = IndexCtx(window_size_r=30)
+    index = indices_ctx.tiered_index
+    data = indices_ctx.data
+    num_elements = indices_ctx.num_labels
+
+    # Create svs index.
+    svs_index = indices_ctx.init_and_populate_svs_index()
+
+    test_logger.info(f"add {indices_ctx.num_labels} vectors to the tiered index one by one")
+
+    # Populate tiered index.
+    vectors = []
+    for i, vector in enumerate(data):
+        index.add_vector(vector, i)
+        vectors.append((i, vector))
+
+    index.wait_for_index()
+
+    test_logger.info(f"Deleting half of the index")
+    # Delete half of the index.
+    for i in range(0, num_elements, 2):
+        index.delete_vector(i)
+        svs_index.delete_vector(i)
+
+    # Wait for all repair jobs to be done.
+    index.wait_for_index(5)
+    test_logger.info(f"Done deleting half of the index")
+    assert index.svs_label_count() >= (num_elements / 2) - indices_ctx.tiered_svs_params.updateTriggerThreshold
+    assert svs_index.index_size() == (num_elements / 2)
+
+    # Create a list of tuples of the vectors that left.
+    vectors = [vectors[i] for i in range(1, num_elements, 2)]
+
+    # Perform queries.
+    num_queries = 10
+    queries = indices_ctx.generate_queries(num_queries=10)
+
+    k = 10
+    correct_tiered = 0
+    correct_svs = 0
+
+    # Calculate correct vectors for each index.
+    # We don't expect svs and tiered svs results to be identical due to the parallel insertion.
+    def calculate_correct(index_labels, keys):
+        correct = 0
+        for label in index_labels[0]:
+            for correct_label in keys:
+                if label == correct_label:
+                    correct += 1
+                    break
+        return correct
+
+    for target_vector in queries:
+        tiered_labels, _ = index.knn_query(target_vector, k)
+        svs_labels, _ = svs_index.knn_query(target_vector, k)
+
+        # Sort distances of every vector from the target vector and get actual k nearest vectors.
+        dists = [(spatial.distance.cosine(target_vector, vec), key) for key, vec in vectors]
+        dists = sorted(dists)
+        keys = [key for _, key in dists[:k]]
+        correct_tiered += calculate_correct(tiered_labels, keys)
+        correct_svs += calculate_correct(svs_labels, keys)
+
+    # Measure recall.
```

Comment:
Can we also confirm that we reclaimed the memory properly after deletion? to validate that svs GC was triggered as expected

Reply:
This test is copied from [HNSW Tiered flow test](https://github.com/RedisAI/VectorSimilarity/blob/main/tests/flow/test_hnsw_tiered.py#L339).
Can you please provide an example, how this check should look like in case of HNSW?
Thank you.

Reply:
I guess we don't have such validation in HNSW tiered equivalent test. Never mind, no need to check it here, we will add a test to the svs GC somewhere else.